src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -675,11 +699,6 @@ private bool TryTakeWithNoTimeValidation([MaybeNullWhen(false)] out T item, int
 
```

Comment:
Can the `CheckDisposed();`  call in line 673 also be removed in favor of an assert? It seems like `IsCompleted` is also making that check.